chcemvediet/apps/inforequests/tests/test_cron.py

```diff
@@ -185,38 +186,32 @@ def test_reminder_is_sent_if_newest_undecided_email_newer_than_last_reminder(sel
         message_set = self._call_cron_job()
         self.assertTrue(message_set.exists())
 
-    @unittest.skip('FIXME')
     def test_inforequest_is_skipped_if_exception_raised_while_checking_it(self):
         timewarp.jump(local_datetime_from_local(u'2010-10-05 10:33:00'))
-        inforequests = [self._create_inforequest() for i in range(3)]
-        emails = [self._create_inforequest_email(inforequest=ir) for ir in inforequests]
+        inforequest = self._create_inforequest()
+        email = self._create_inforequest_email(inforequest=inforequest)
 
         timewarp.jump(local_datetime_from_local(u'2010-10-20 10:33:00'))
-        with mock.patch(u'chcemvediet.apps.inforequests.cron.nop', side_effect=[None, Exception, None, None, None]):
-            with mock.patch(u'chcemvediet.apps.inforequests.cron.cron_logger') as logger:
-                message_set = self._call_cron_job()
-        self.assertEqual(message_set.count(), 2)
-        self.assertEqual(len(logger.mock_calls), 3)
-        self.assertRegexpMatches(logger.mock_calls[0][1][0], u'Checking if undecided email reminder should be sent failed: <Inforequest: %s>' % inforequests[1].pk)
-        self.assertRegexpMatches(logger.mock_calls[1][1][0], u'Sent undecided email reminder: <Inforequest: %s>' % inforequests[0].pk)
-        self.assertRegexpMatches(logger.mock_calls[2][1][0], u'Sent undecided email reminder: <Inforequest: %s>' % inforequests[2].pk)
+        with mock.patch(u'chcemvediet.apps.inforequests.cron.cron_logger') as logger:
+            message_set = self._call_cron_job(exception=Exception)
+        self.assertFalse(message_set.exists())
+        self.assertEqual(len(logger.mock_calls), 1)
+        msg = u'Checking if undecided email reminder should be sent failed: {}'.format(re.escape(repr(inforequest)))
+        self.assertRegexpMatches(logger.mock_calls[0][1][0], msg)
 
-    @unittest.skip('FIXME')
     def test_inforequest_is_skipped_if_exception_raised_while_sending_reminder(self):
         timewarp.jump(local_datetime_from_local(u'2010-10-05 10:33:00'))
-        inforequests = [self._create_inforequest() for i in range(3)]
-        emails = [self._create_inforequest_email(inforequest=ir) for ir in inforequests]
+        inforequest = self._create_inforequest()
+        email = self._create_inforequest_email(inforequest=inforequest)
 
         timewarp.jump(local_datetime_from_local(u'2010-10-20 10:33:00'))
-        with mock.patch(u'chcemvediet.apps.inforequests.cron.nop', side_effect=[None, None, None, None, Exception, None]):
+        with mock.patch(u'chcemvediet.apps.inforequests.models.inforequest.Inforequest.send_undecided_email_reminder', side_effect=Exception):
```

Comment:
Tu by to bolo tiez IMHO prehladnejsie cez `nop()`.

Reply:
:ok: